### PR TITLE
Add a performance.measure polyfill

### PR DESCRIPTION
**Pull request by Velveeta**
Adding a polyfill for performance.measure, as well as removing some cruft that was [pointed out in the previous PR](https://github.com/bazaarvoice/bv-ui-core/pull/83#discussion_r160748671), and adding test coverage.

### CODE REVIEW

File: lib/performance/measure.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+function polyfillImplementation (name, startMark, endMark) {
+  // Validation: a measure name must be provided
+  if (arguments.length < 1) {
+    throw new TypeError('Failed to execute \'measure\' on \'Performance\': 1 argument required, but only 0 present.');
+  }
+
+  var marks = perfGetEntriesByType('mark');
+  var filteredMarks;
+
+  // if no startMark was passed, used
```
Comment (tnunamak) [inline lib/performance/measure.js]: Looks like some of this comment was lef

Comment (Velveeta) [reply lib/performance/measure.js]: ahhhhh ye

File: lib/performance/measure.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  var startTime = 0;
+  var endTime = perfNow();
+
+  if (startMark !== undefined) {
+    filteredMarks = marks.filter(function (entry) {
+      return entry.entryName === startMark;
+    });
+    if (!filteredMarks.length) {
+      throw new Error('Failed to execute \'measure\' on \'Performance\': The mark \'foosdfsdf\' does not exist.');
```
Comment (tnunamak) [inline lib/performance/measure.js]: foosdfsdf

File: lib/performance/measure.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    // use the most recent mark with the matching name
+    startTime = filteredMarks[filteredMarks.length-1].startTime;
+  }
+
+  if (endMark !== undefined) {
+    filteredMarks = marks.filter(function (entry) {
+      return entry.entryName === endMark;
+    });
+    if (!filteredMarks.length) {
+      throw new Error('Failed to execute \'measure\' on \'Performance\': The mark \'foosdfsdf\' does not exist.');
```
Comment (tnunamak) [inline lib/performance/measure.js]: More foosdfsdf

File: lib/performance/measure.js (hunk 2 above)
Comment (Velveeta) [reply lib/performance/measure.js]: Ha! whoops :D

**Comment by Velveeta** [comment]
Ok, pushed changes to fix those items, good catches :)

**Pull request merged**